src/main/kotlin/no/nav/helse/spion/domene/ytelsesperiode/repository/PostgresYtelsesperiodeRepository.kt

```diff
@@ -34,14 +41,24 @@ class PostgresYtelsesperiodeRepository(val ds: DataSource, val mapper: ObjectMap
             AND data -> 'periode' ->> 'tom' = ?;"""
 
 
-    override fun getYtelserForPerson(identitetsnummer: String, virksomhetsnummer: String): List<Ytelsesperiode> {
+    override fun getYtelserForPerson(identitetsnummer: String, virksomhetsnummer: String, periode: Periode?): List<Ytelsesperiode> {
         ds.connection.use { con ->
-
-            val res = con.prepareStatement(getByPersonAndArbeidsgiverStatement).apply {
-                setString(1, identitetsnummer)
-                setString(2, virksomhetsnummer)
-            }.executeQuery()
             val resultList = ArrayList<Ytelsesperiode>()
+            val res = periode?.let {
+                con.prepareStatement(getByPersonAndArbeidsgiverInPeriodStatement).apply {
+                    setString(1, identitetsnummer)
+                    setString(2, virksomhetsnummer)
+                    setString(3, periode.fom.toString())
+                    setString(4, periode.tom.toString())
+                    setString(5, periode.tom.toString())
+                    setString(6, periode.fom.toString())
```

Comment:
Når du får så mange og gjenbrukte parametere, kan du bruke NamedParameters:

`String query = "select * from people where (first_name = :name or last_name 
= :name) and address = :address");
NamedParameterStatement p = new NamedParameterStatement(con, query);
p.setString("name", name);
p.setString("address", address);`

Reply:
Det blir jo mye penere! Men vi bruker ikke sql2o i Spion(enda), er dette grunn nok til å dra det inn? Det kan vel erstatte deler av java.sql.* 🤔

Reply:
Det var ikke trivielt å bytte til sql2o, så jeg gjør ikke noe med det akkurat nå. Om vi får fler spørringer med mange parametre kan vi alltids bytte da 😄